src/tensor/keys_to_properties.rs

```diff
@@ -197,19 +195,23 @@ fn merge_blocks_along_properties(
     // the corresponding data
     let mut property_ranges = Vec::new();
     for (new_property, block) in blocks_to_merge {
-        let size = block.values().properties.count();
+        if block.values().properties.is_empty() {
+            // no properties, ignore this block
+            property_ranges.push(None);
+            continue;
+        }
 
         let mut first = new_property.clone();
         first.extend_from_slice(&block.values().properties[0]);
 
-        // we can lookup only the `first` new property here, since all blocks
-        // are checked to have the same properties, which makes the new
+        // we can lookup only the `first` new property here, since the new
```

Comment:
I never capitalize comments inside code, not sure why =) 

Might leave this one as is since all the other comments in the file are also not capitalized

Reply:
Fine for me